### PR TITLE
Add initial mkdocs documentation website

### PR DESCRIPTION
**Pull request by DaniruKun**
As discussed with @axelson on Slack, this is the first version of the ElixirLS documentation website built using [Mkdocs](https://www.mkdocs.org)

At the moment, it contains everything a user requires to learn how to use and install ElixirLS for any supported editor, as well as intuitive navigation, links and universal search.

image

The GitHub Actions for automatic publishing of the website to Github Pages will be added in a future PR, as well as more screenshots and GIFs showcasing the features.

At the moment, the website can be manually served using `mkdocs serve` (for development) and built using `mkdocs build`.

### CODE REVIEW

**Review by axelson — COMMENTED** [review]
Thanks for this! The site will be a nice improvement for users.

And (as mentioned in Slack) I like keeping it all in one repository because then a PR can update the website as well.

As part of the this transition I'm thinking maybe should start using a `dev` branch, and only merge `master` when making a release. But I'll start another issue for that.

For this PR can you:
- add `site` to `.gitignore`
- write up some instructions on building the documentation website locally in `DEVELOPMENT.md`

Thanks!

**Comment by DaniruKun** [comment]
@axelson added the changes you requested. Regarding the automated publishing of changes to the site - it looks like there is a new Github Action for this that greatly simplifies this, should be very easy to setup: https://github.com/jamesIves/github-pages-deploy-action

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -26,3 +26,15 @@ Publish the draft release
 If you're debugging a running server than `IO.inspect` is a good approach, any messages you create with it will be sent to your LSP client as a log message
 
 To debug in tests you can use `IO.inspect(Process.whereis(:user), message, label: "message")` to send your output directly to the group leader of the test process.
+
+# Documentation website
+
+The documentation website is built using the [Mkdocs](https://www.mkdocs.org) static website generator. The content is written in Markdown format in the directory [docs](./docs) and is configured via the [mkdocs.yml](./mkdocs.yml) file.
```
Comment (axelson) [inline DEVELOPMENT.md:32]: Do you have a list of any additional packages needed? I've partially gotten this running locally and it looks like you need at least `pip install mkdocs-material`, are there any others? Can we add a note about them?

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -26,3 +26,15 @@ Publish the draft release
 If you're debugging a running server than `IO.inspect` is a good approach, any messages you create with it will be sent to your LSP client as a log message
 
 To debug in tests you can use `IO.inspect(Process.whereis(:user), message, label: "message")` to send your output directly to the group leader of the test process.
+
+# Documentation website
```
Comment (axelson) [inline DEVELOPMENT.md:30]: Is there a minimum python version needed for this?

**Comment by axelson** [comment]
That new GitHub action looks like it'll be good. Let's add it to `.github/workflows/release-asset.yml` as a new job

File: DEVELOPMENT.md (hunk 1 above)
Comment (DaniruKun) [reply DEVELOPMENT.md:32]: Added the required packages in the doc.

File: DEVELOPMENT.md (hunk 2 above)
Comment (DaniruKun) [reply DEVELOPMENT.md:30]: At least Python 3, mentioned in the doc now. As long as Mkdocs works.

**Comment by DaniruKun** [comment]
@axelson updated with your comments. I set Github Actions to deploy the build website into the branch `mkdocs`. This is the one that you have to configure Github to present from.

**Review by axelson — APPROVED** [review]
Thanks again for this! :heart:

**Comment by DaniruKun** [comment]
I think next we can add the link to the site somewhere in the repo description, or maybe also in the org home. It might also be worth revisiting the README and removing a bunch of stuff that was migrated to the site anyway, and just leave a quick QuickStart guide and a link to the site.

**Pull request merged**